templates/terraform/resource.erb

```diff
@@ -309,14 +358,38 @@ func resource<%= resource_name -%>Import(d *schema.ResourceData, meta interface{
 		return nil, fmt.Errorf("Error constructing id: %s", err)
 	}
 	d.SetId(id)
+  <%= lines(compile(object.custom_code.post_import)) if object.custom_code.post_import -%>
 
 	return []*schema.ResourceData{d}, nil
+<% end -%>
 }
 
 <% properties.each do |prop| -%>
+<% if object.custom_code.custom_flatten.key? prop.name -%>
+<%= lines(compile_template(object.custom_code.custom_flatten[prop.name],
+                           prefix: resource_name,
+                           prop: prop)) -%>
+<% else -%>
 <%= lines(build_flatten_method(resource_name, prop), 1) -%>
 <% end -%>
+<% end -%>
 
 <% settable_properties.each do |prop| -%>
+<% if object.custom_code.custom_expand.key? prop.name -%>
+  <%= lines(compile_template(object.custom_code.custom_expand[prop.name],
+                           prefix: resource_name,
+                           prop: prop))-%>
+<% else -%>
 <%= lines(build_expand_method(resource_name, prop), 1) -%>
 <% end -%>
+<% end -%>
+
+<% properties.each do |prop| -%>
+  <% if object.custom_code.decoder.key? prop.name -%>
+func customDecoder<%= resource_name -%><%= titlelize_property(prop) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
```

Comment:
any reason the encoders are at the top and the decoders at the bottom of the file? if not, let's move the encoders down here too

Reply:
Sort of felt more right to have all the global stuff up top and all the per-prop stuff down here ... but your way's better, moved it.